# Remote Management Request
temoto_core/RMPRequest rmp

# Name of the component (not required)
string component_name

# Type of the requested component (required).
string component_type

# Name of the ROS package where component is located. (optional)
# If the package name is left undefined
# then the first known entry under the specified type is executed
string package_name

# name of the executable (optional)
# When undefined, first defined entry is used.
string executable

bool use_only_local_components

# Output topics. Each element is a pair of strings where the 
# first element (key) describes the type of the topic and the
# second element (value) describes the name of the topic itself.
# Key is required but if the value is left empty, a default
# topic name is used (indicated by the component info file)
# by the component manager
diagnostic_msgs/KeyValue[] output_topics

# Same idea applies here as it did with the output_topics
diagnostic_msgs/KeyValue[] input_topics

# Same idea applies here as it did with the output_topics
diagnostic_msgs/KeyValue[] required_parameters

---

# Remote Management Response
temoto_core/RMPResponse rmp

# name of the package
string package_name

# name of the executable/launchable/runnable
string executable

# Actual output topics
diagnostic_msgs/KeyValue[] output_topics

diagnostic_msgs/KeyValue[] input_topics

diagnostic_msgs/KeyValue[] required_parameters
